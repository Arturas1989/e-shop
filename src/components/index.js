export { Header } from './Header/Header';
export { NavigationMenu } from './Header/Menu/NavigationMenu/NavigationMenu';
export { LoginMenu } from './Header/Menu/LoginMenu/LoginMenu';
export { Hamburger } from './Header/Hamburger/Hamburger';
export { CloseMenu } from './Header/Menu/NavigationMenu/CloseMenu';
export { LoginLink } from './Header/Menu/LoginMenu/LoginLink';

export { SearchBar } from './SearchBar/SearchBar';
export { InputBar } from './SearchBar/InputBar';

export { Footer } from './Footer/Footer';

export { PrimaryButton } from './Buttons/PrimaryButton/PrimaryButton';
export { SecondaryButton } from './Buttons/SecondaryButton/SecondaryButton';

export { Rating } from './ProductCard/Rating/Rating';
export { BestSeller } from './ProductCard/BestSeller/BestSeller';
export { Heading } from './Heading/Heading';
