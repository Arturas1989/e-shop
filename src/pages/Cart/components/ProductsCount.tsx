import { useCartContext } from "../../../contexts/cartContext";
import { useCartProducts } from "../../../hooks";

export const ProductsCount = () => {
  const {cart} = useCartContext()!;
  const [, products] = useCartProducts(cart);

  return (
    <span>({products?.length})</span>
  )
};
