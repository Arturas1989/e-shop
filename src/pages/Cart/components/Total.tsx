import { Product } from "../../../types";

type TotalProps = {
  products: Product[] | null;
};

export const Total = ({ products }: TotalProps) => {
  let total;
  if(products)total = products.reduce((acc, product) => acc + product.price * product.quantity!, 0);
  return (
    <div className="total-row">
      <h2>Total amount:</h2>
      <span>{total}$</span>
    </div>
  )
};

