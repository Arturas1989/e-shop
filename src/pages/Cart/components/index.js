export { ProductsCount } from './ProductsCount';
export { CartCard } from './Cartcard/CartCard';
export { CartProducts } from './CartProducts';
