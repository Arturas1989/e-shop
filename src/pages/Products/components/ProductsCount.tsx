import { Loading } from "../../../components";
import { useProductsContext } from "../../../contexts/productContext";

export const ProductsCount = () => {
  const {products, isProductsLoading} = useProductsContext()!
  return (
    isProductsLoading ? <h2>All products ({products?.length})</h2> : <Loading />
  )
};
