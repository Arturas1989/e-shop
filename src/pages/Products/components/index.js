export { FilterOpenButton } from './Filter/FilterOpenButton';
export { ProductsCount } from './ProductsCount';
export { Filter } from './Filter/Filter';
export { InputGroup } from './Filter/InputGroup';
