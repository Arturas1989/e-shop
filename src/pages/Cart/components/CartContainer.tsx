import { SecondaryButton } from "../../../components";
import { Product } from "../../../types";
import { CartCard } from "./Cartcard/CartCard";
import { Total } from "./Total";

type CartContainerProps = {
  products: Product[] | null;
};

export const CartContainer = ({ products }: CartContainerProps) => {
  return (
    <>
      {products?.map((product) => (
        <CartCard key={product.id} product={product} />
      ))}
      <Total products={products} />
      <SecondaryButton>
        Place order<i className="bi bi-arrow-right"></i>
      </SecondaryButton>
    </>
  );
};
