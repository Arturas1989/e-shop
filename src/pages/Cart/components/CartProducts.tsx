import { Loading, SecondaryButton } from '../../../components';
import { useCartContext } from '../../../contexts/cartContext';
import { useCartProducts } from '../../../hooks';
import { CartCard } from './Cartcard/CartCard';
import { Total } from './Total';

type CartProductsProps = {
  name?: string;
};

export const CartProducts = ({ name }: CartProductsProps) => {
  const { cart } = useCartContext()!;
  const [isLoading, products] = useCartProducts(cart);
  console.log(products);
  return (
    <section className="cards">
      {isLoading ? (
        <Loading />
      ) : (
        <>
          {products?.map((product) => (
            <CartCard key={product.id} product={product} />
          ))}
          <Total products={products} />
          <SecondaryButton>
            Place order<i className="bi bi-arrow-right"></i>
          </SecondaryButton>
        </>
      )}
    </section>
  );
};
