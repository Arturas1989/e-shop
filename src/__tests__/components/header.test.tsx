import renderer from 'react-test-renderer';
import { fireEvent, screen, render } from '@testing-library/react';
import { Header } from '../../components';
import { BrowserRouter } from 'react-router-dom';
import { initializeApp } from 'firebase/app';
import { firebaseConfig } from '../../db/firebase';

describe('header components renders', () => {
  test('renders correctly', () => {
    initializeApp(firebaseConfig);
    const tree = renderer
      .create(
        <BrowserRouter>
          <Header />
        </BrowserRouter>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('hamburger renders correctly and items on click changes correctly', () => {
    render(
      <BrowserRouter>
        <Header />
      </BrowserRouter>
    );
    const hamburger = screen.getByTestId('hamburger');
    expect(hamburger).toBeInTheDocument();

    fireEvent.click(hamburger);
    const closeIcon = screen.getByText('close menu');
    expect(closeIcon).toBeInTheDocument();

    const menu = screen.getByTestId('menu');
    expect(menu).toBeInTheDocument();

    fireEvent.click(hamburger);
    const hamburgerIcon = screen.getByText('menu');
    expect(hamburgerIcon).toBeInTheDocument();
  });
});
