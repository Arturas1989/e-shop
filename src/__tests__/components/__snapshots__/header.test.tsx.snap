// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`header components renders renders correctly 1`] = `
<header
  className="search-not-visible"
  data-testid="header"
  id="header"
>
  <div
    className="navbar-container"
  >
    <nav
      className="navbar"
    >
      <div
        className="left-nav"
      >
        <div
          className="hamburger"
          data-testid="hamburger"
          onClick={[Function]}
        >
          <div
            className="hidden"
          >
            <i
              className="bi bi-x-diamond-fill"
            />
            <p>
              close menu
            </p>
          </div>
          <div
            className="visible"
          >
            <i
              className="bi bi-list"
            />
            <p>
              menu
            </p>
          </div>
        </div>
        <a
          href="/"
          onClick={[Function]}
        >
          <img
            alt="e-shop-logo"
            src="e-shop-logo.png"
          />
          <p>
            e-shop
          </p>
        </a>
      </div>
      <div
        className="middle-nav"
      >
        <div>
          <a
            aria-current="page"
            className="active"
            href="/"
            onClick={[Function]}
          >
            Home
          </a>
        </div>
        <div>
          <a
            className=""
            href="/products"
            onClick={[Function]}
          >
            Products
          </a>
        </div>
      </div>
      <div
        className="right-nav"
      >
        <i
          className="bi bi-gear-wide-connected"
        />
        <i
          className="bi bi-search"
          data-testid="search-icon"
          onClick={[Function]}
        />
        <i
          className="bi bi-cart-fill"
        >
          <div
            className="cart-items"
          >
            0
          </div>
        </i>
        <i
          className="bi bi-person-circle"
        />
      </div>
    </nav>
  </div>
  <div
    className="menu notVisible"
    data-testid="menu"
  >
    <nav>
      <a
        aria-current="page"
        className="active"
        href="/"
        onClick={[Function]}
      >
        Home
      </a>
      <a
        className=""
        href="/cart"
        onClick={[Function]}
      >
        Cart
      </a>
      <a
        className=""
        href="/products"
        onClick={[Function]}
      >
        Products
      </a>
      <i
        className="bi bi-search"
        data-testid="search-icon2"
        onClick={[Function]}
      />
    </nav>
  </div>
</header>
`;
